docs/docs-content/automation/palette-sdk/list-clusters-sdk.md

```diff
@@ -0,0 +1,174 @@
+---
+sidebar_label: "List Clusters with Palette Go SDK"
+title: "List Clusters with Palette Go SDK"
+description: "Learn how to use the Palette GO SDK to list your Palette host clusters."
+hide_table_of_contents: false
+sidebar_position: 10
+tags: ["palette-sdk", "go"]
+---
+
+The Palette Go Software Developer Toolkit (SDK) enables Go developers to interact with Palette services. This guide
+explains how to install and use the SDK to list the active host clusters in your Palette environment.
+
+## Prerequisites
+
+- [Go](https://go.dev/doc/install) version 1.22.5 or later.
+
+- A Palette account with [Tenant Admin](../../tenant-settings/tenant-settings.md) access.
+
+- A Palette API key. Refer to the [Create API Key](../../user-management/authentication/api-key/create-api-key.md) page
+  for instructions on how to create an API key.
+
+## List Palette Clusters
+
+1.  Open a terminal window and create a folder for you Go repository.
+
+    ```bash
+    mkdir sdk-example
+    ```
+
+2.  Navigate to the created folder.
+
+    ```bash
+    cd sdk-example
+    ```
+
+3.  Initialize your Go module and enable dependency tracking for your code by issuing the following command. This will create a `go.mod` file in the
+    current folder.
+
+    ```bash
+    go mod init example/list-clusters
+    ```
+
+4.  Download the required Palette SDK modules. The command below retrieves the modules and records their dependencies in
+    the `go.mod` file.
+
+    ```bash
+    go get github.com/spectrocloud/palette-sdk-go/api/models
+    go get github.com/spectrocloud/palette-sdk-go/client
+    ```
+
+5.  Create a source file named `main.go`.
+
+        ```bash
+        touch main.go
+        ```
+
+6.  Open the `main.go` file in a text editor of your choice and paste the content below.
+
+    ```go
+    package main
+
+    import (
+        "fmt"
+        "os"
+        "github.com/spectrocloud/palette-sdk-go/api/models"
+        "github.com/spectrocloud/palette-sdk-go/client"
+    )
+
+    func main() {
+        var host, apiKey, projectUid, scope string
+
+        // Parse command line arguments
+
+        numArgs := len(os.Args)
+        if numArgs < 3 || numArgs > 4 {
+            fmt.Println("Usage: main <host> <apiKey> [projectUid]")
+            os.Exit(1)
+        }
+        switch numArgs {
+        case 3:
+            host, apiKey = os.Args[1], os.Args[2]
+            scope = "tenant"
+        case 4:
+            host, apiKey, projectUid = os.Args[1], os.Args[2], os.Args[3]
+            scope = "project"
+        }
+
+        // Initialize a Palette client
```

Comment:
Use `os.getenv`. to get the ENV variables with the credentials and Palette URI.  That's the more idiomatic pattern in Go.

https://pkg.go.dev/os#Getenv

Reply:
I'd be happy to update the example in the SDK repo to use env vars. Agreed with @caroldelwing that they should be consistent.